### PR TITLE
bunch of opinionated improvements :)

### PR DESCRIPTION
**Pull request by alexweber**
Hey @joaogarin, these are the bulk of the changes that I made in order to get my project going.

Highlights
- Electron 1.4.3
- SCSS support for components
- External templates for components
- Support for webfonts (not demo'ed yet)
- Home & Login components in their own directories (was getting unwieldy with new html and scss files)

Caveats
- I absolutely couldn't get this setup working using Node < 6
- My project still used Webpack 1.x and, with that, it's not actually necessary to do the ugly `require` for html templates.  I couldn't get it working any other way with Webpack 2.x (I suspect it's an issue with the angular2-template-loader)
- The `require` is _still_ needed for styles because of [this issue](https://github.com/webpack/style-loader/issues/123)

I realize it's a big change and super opinionated so I don't expect you to merge it as-is... either way, I wanted to share it and get your thoughts.

I personally don't mind Node 6 nor the requires so I'm happy with it but Webpack is largely still unfamiliar territory for me so I could probably do better :)  Thanks!

### CODE REVIEW

**Comment by joaogarin** [comment]
I definitely want to add some of these stuff. The one I am a bit cautious about is having to have node 6 for this..thats not something I really like. I would basically put some of these together and see where this node 6 thing starts to be a problem.

File: package.json [hunk 1]
```diff
@@ -62,24 +65,29 @@
...
     "typings": "^1.3.2",
+    "url-loader": "^0.5.7",
     "webpack": "2.1.0-beta.22",
-    "webpack-dev-server": "^2.1.0-beta.2",
-    "source-map-loader": "^0.1.5"
+    "webpack-dev-server": "^2.1.0-beta.2"
   },
   "engines": {
-    "node": ">= 4.2.1 <= 5",
+    "node": ">= 6",
```
Comment (joaogarin) [inline package.json:90]: this seems to be to me the only real issue with the PR. I dont want to force node 6. Do you know where this starts to become a problem? Why is that needed? I seem to be able to run this on node 5..

File: src/app/components/login/login.component.ts [hunk 2]
```diff
@@ -8,21 +8,17 @@ import { Router } from '@angular/router';
...
 @Component({
     selector: 'ae-login',
-    template: `
-    <div>
-        Authenticate
-        <button *ngIf='!authenticated' (click)='authenticate()'>Authenticate with Github</button>
-    </div>
-    `
+    template: require('./login.component.html').toString(),
+    styles: [require('./login.component.scss').toString()],
```
Comment (joaogarin) [inline src/app/components/login/login.component.ts:21]: Maybe you can try this Maybe you can try this? https://github.com/AngularClass/angular2-webpack-starter/blob/master/src/app/home/home.component.ts#L17 Putting the Styles in an array in StylesUrl

File: src/app/components/home/home.component.ts [hunk 3]
```diff
@@ -8,16 +8,12 @@ import { FormControl, FormGroup } from '@angular/forms';
...
 @Component({
   selector: 'ae-home',
-  template: `
-    <div [formGroup]="messageForm">
-       <h1>{{messageForm.controls['messageText'].value}}</h1>
-       <input type="text" formControlName="messageText"/>
-    </div>
-    `
+  template: require('./home.component.html').toString(),
+  styles: [require('./home.component.scss').toString()],
```
Comment (joaogarin) [inline src/app/components/home/home.component.ts:16]: Maybe you can try this Maybe you can try this? https://github.com/AngularClass/angular2-webpack-starter/blob/master/src/app/home/home.component.ts#L17 Putting the Styles in an array in StylesUrl

**Review by joaogarin — COMMENTED** [review]
Hello,

Looks good, I am just thinking about how to get rid of the node 6 thing. Regarding the URL's I think the require might not be needed if we use StylesUrl and templateUrl like on https://github.com/AngularClass/angular2-webpack-starter/blob/master/src/app/home/home.component.ts#L17 . that will be also AOT friendly and the same system being used currently for the CLI. I think we could use that.

**Review by joaogarin — COMMENTED** [review]
Honestly I have tested here, eveything looks good to me. but I think we dont need the node 6 version. I am on node 5 and this works for me.

I would suggest even changing to : 

``` javascript
 "node": ">= 4.2.1",
```

Other than that on my part is good to go. I am getting a warning on electron regarding this https://github.com/electron/electron/issues/4420 but this seems to be electron related, I dont see anything in the app that would be creating this issue so I thinkg with that fix we are good to go.

**Comment by alexweber** [comment]
@joaogarin Awesome, thanks for the detailed feedback!  PS - I had a bunch of issues upgrading further to Webpack 2 beta 25 (total config file change) so I get why you went with 22 now :)

Also, my project uses node-sqlite3, maybe that's where node 6 might be coming in, maybe i just need to npm cache clear or something. I'll be online in a few hours to discuss more thanks again! :)

**Comment by joaogarin** [comment]
No problem, I can also migrate this now. So I merge but I just revert the part of the node in the package json. because everything else would work I guess, I can also later on try the templateURls and StylesUrls like I mentioned up there. But if thats okay with you I can merge this now and do the changes on a later stage.

**Comment by alexweber** [comment]
@joaogarin of course it's ok with me!  I'm just happy to contribute and help figure out Angular2 + Electron stuff :)

PS - Main.js uses some ES6 stuff that will probably not work on Node 4.x  Probably just a matter of changing the shorthand function syntax back for the most part
